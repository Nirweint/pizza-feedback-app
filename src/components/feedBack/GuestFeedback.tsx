import React from "react";
import { Box, Button, Paper, Rating, Typography } from "@mui/material";
import {FeedbackType} from "../../types";
import {getLocalStorageState, setLocalStorageState} from "../../localStorage";

type GuestFeedbackPropsType = {
  currentGuestFeedback: FeedbackType;
  setOpenModal: (value: boolean) => void;
};

export const GuestFeedback = ({
  currentGuestFeedback,
  setOpenModal
}: GuestFeedbackPropsType) => {
  const { name, phone, rating, comment } = currentGuestFeedback;

  const feedbacksFromLocalStorage = getLocalStorageState<FeedbackType[]>(
    "feedback",
    []
  );

  const handleDeleteClick = () => {
    const newFeedbackData = feedbacksFromLocalStorage.filter(
      (guest) => guest.name !== name
    );
    setLocalStorageState("feedback", newFeedbackData);
    setOpenModal(false);
  };

  return (
    <div>
      <Paper elevation={3}>
        <Box p={2} sx={{ display: "flex", flexDirection: "column" }}>
          <Box sx={{ display: "flex", justifyContent: "space-between" }}>
            <Typography>Name</Typography>
            <Button onClick={handleDeleteClick} color={"error"}>
              delete
            </Button>
          </Box>
          <Typography variant={"h4"}>{name}</Typography>
          <Box sx={{ margin: "10px 0" }}>
            <Rating name="read-only" value={rating} readOnly />
          </Box>
          <Box sx={{ margin: "0 0 10px 0" }}>
            <Typography variant={"h5"} sx={{ alignSelf: "flex-start" }}>
              PHONE
            </Typography>
            <Typography>{phone}</Typography>
          </Box>
          <Box sx={{ margin: "0 0 10px 0" }}>
            <Typography variant={"h5"} sx={{ alignSelf: "flex-start" }}>
              COMMENT
            </Typography>
            <Typography>{comment}</Typography>
          </Box>
        </Box>
      </Paper>
    </div>
  );
};
