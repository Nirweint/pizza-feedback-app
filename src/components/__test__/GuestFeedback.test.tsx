import React from 'react';
import renderer from 'react-test-renderer';
import {GuestFeedback} from "../GuestFeedback";

const mockData = {
  name: 'Alex',
  rating: 4,
  phone: '22222',
  comment: 'Hello'
}

test('GuestFeedback renders correctly', () => {
  const component = renderer.create(<GuestFeedback currentGuestFeedback={mockData} setOpenModal={() => {}}/>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});