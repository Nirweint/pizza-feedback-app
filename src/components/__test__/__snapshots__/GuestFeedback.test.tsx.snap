// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestFeedback renders correctly 1`] = `
<div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
  >
    <div
      className="MuiBox-root css-1siwe0b"
    >
      <div
        className="MuiBox-root css-gg4vpm"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Name
        </p>
        <button
          className="MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-10fi992-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          delete
        </button>
      </div>
      <h4
        className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        Alex
      </h4>
      <div
        className="MuiBox-root css-1tx38fa"
      >
        <span
          aria-label="4 Stars"
          className="MuiRating-root MuiRating-sizeMedium MuiRating-readyOnly css-ryrseu-MuiRating-root"
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          role="img"
        >
          <span>
            <span
              className="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              className="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              className="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              className="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarBorderIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
      <div
        className="MuiBox-root css-dkdl5e"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 css-1vhf2cv-MuiTypography-root"
        >
          PHONE
        </h5>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          22222
        </p>
      </div>
      <div
        className="MuiBox-root css-dkdl5e"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 css-1vhf2cv-MuiTypography-root"
        >
          COMMENT
        </h5>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Hello
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
  >
    <div
      className="MuiBox-root css-1siwe0b"
    >
      <div
        className="MuiBox-root css-gg4vpm"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Name
        </p>
        <button
          className="MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-10fi992-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          delete
        </button>
      </div>
      <h4
        className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        Alex
      </h4>
      <div
        className="MuiBox-root css-1tx38fa"
      >
        <span
          aria-label="4 Stars"
          className="MuiRating-root MuiRating-sizeMedium MuiRating-readyOnly css-ryrseu-MuiRating-root"
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          role="img"
        >
          <span>
            <span
              className="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              className="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              className="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              className="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarBorderIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
      <div
        className="MuiBox-root css-dkdl5e"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 css-1vhf2cv-MuiTypography-root"
        >
          PHONE
        </h5>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          22222
        </p>
      </div>
      <div
        className="MuiBox-root css-dkdl5e"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 css-1vhf2cv-MuiTypography-root"
        >
          COMMENT
        </h5>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Hello
        </p>
      </div>
    </div>
  </div>
</div>
`;
